deployments/container/Dockerfile

```diff
@@ -44,10 +44,28 @@ ARG VERSION="N/A"
 ARG GIT_COMMIT="unknown"
 RUN make PREFIX=/artifacts cmds
 
-FROM nvidia/cuda:12.6.0-base-ubi8
+# We use the ubi8-minimal image as a reference image in removing unneeded dependencies.
+FROM redhat/ubi8-minimal:latest AS minimal
 
-# Remove CUDA libs(compat etc) in favor of libs installed by the NVIDIA driver
-RUN dnf remove -y cuda-*
+RUN rpm -qa --queryformat='^%{NAME}-\[0-9\].*\.%{ARCH}$\n' | sort -u > /tmp/package-names.minimal
+RUN rpm -qa | sort -u > /tmp/package-list.minimal
```

Comment:
```suggestion
RUN rpm -qa --queryformat='^%{NAME}-\[0-9\].*\.%{ARCH}$\n' | sort -u > /tmp/package-names.minimal && \
  rpm -qa | sort -u > /tmp/package-list.minimal
```

Reply:
Same comment as in https://github.com/NVIDIA/k8s-test-infra/pull/61#discussion_r1674132513 ... I would like to keep these separate.